src/braket/ahs/analog_hamiltonian_simulation.py

```diff
@@ -63,6 +69,98 @@ def _hamiltonian_to_ir(self) -> ir.Hamiltonian:
             drivingFields=terms["driving_fields"], shiftingFields=terms["shifting_fields"]
         )
 
+    def discretize(self, device) -> AnalogHamiltonianSimulation:
+        """ Creates a new AnalogHamiltonianSimulation with all numerical values represented
+            as Decimal objects with fixed precision based on the capabilities of the device.
+
+            Args:
+                device (AwsDevice): The device for which to discretize the program.
+
+            Returns:
+                AnalogHamiltonianSimulation: A discretized version of this program.
+
+            Raises:
+                DiscretizeError: If unable to discretize the program.
+        """
+
+        required_action_schema = 'braket.ir.ahs.program'
+        if (
+            (required_action_schema not in device.properties.action)
+            or (device.properties.action[required_action_schema].actionType != required_action_schema)
+            ):
+            raise DiscretizationError(f"AwsDevice {device} does not accept {required_action_schema} action schema." )
+
+        # Gather resolution values
+
+        register_position_resolution = device.properties.paradigm.lattice.geometry.positionResolution
+        driving_parameters = device.properties.paradigm.rydberg.rydbergGlobal
+        driving_field_resolutions = {
+            'amplitude': {
+                'time': driving_parameters.timeResolution,
+                'value': driving_parameters.rabiFrequencyResolution
+            },
+            'detuning': {
+                'time': driving_parameters.timeResolution,
+                'value': driving_parameters.detuningResolution
+            },
+            'phase': {
+                'time': driving_parameters.timeResolution,
+                'value': driving_parameters.phaseResolution
+            }
+        }
+        shifting_parameters = device.properties.paradigm.rydberg.rydbergLocal
+        shifting_field_resolutions = {
+            'magnitude': {
+                'time': shifting_parameters.timeResolution,
+                'value': shifting_parameters.commonDetuningResolution,
+                'pattern': shifting_parameters.localDetuningResolution
+            }
+        }
+
+        # Discretize register
+
+        try:
+            discretized_register = self.register.discretize(register_position_resolution)
+        except:
+            raise DiscretizationError(f'Failed to discretize register {self.register}')
+
+        # Discretize Hamiltonian
+
+        discretized_hamiltonian = Hamiltonian()
+        for idx, term in enumerate(self.hamiltonian):
+            if isinstance(term, DrivingField):
```

Comment:
Why did you decide not to just call each term with "discretize" and the device parameters, and let the field discretize itself?